test/e2e/specs/site-editor/template-revert.spec.js

```diff
@@ -0,0 +1,272 @@
+		await admin.visitSiteEditor();
+	} );
+
+	test( 'should delete the template after saving the reverted template', async ( {
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		await page.click( 'role=button[name="Show template details"]' );
+
+		// The revert button isn't visible anymore.
+		await expect(
+			page.locator( 'role=menuitem[name=/Clear customizations/i]' )
+		).not.toBeVisible();
+	} );
+
+	test( 'should show the original content after revert', async ( {
+		templateRevertUtils,
+	} ) => {
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfter );
+	} );
+
+	test( 'should show the original content after revert and page reload', async ( {
+		admin,
+		templateRevertUtils,
+	} ) => {
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+		await admin.visitSiteEditor();
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfter );
+	} );
+
+	test( 'should show the edited content after revert and clicking undo in the header toolbar', async ( {
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		// Revert template and check state.
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+		const contentAfterSave = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).not.toEqual( contentAfterSave );
+
+		// Undo revert and check state again.
+		await page.click( 'role=button >> text="Undo"' );
+		const contentAfterUndo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfterUndo );
+	} );
+
+	test( 'should show the edited content after revert and clicking undo in the notice', async ( {
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		const snackBar = page.locator(
+			'role=button[name="Dismiss this notice"]'
+		);
+
+		await snackBar.locator( 'role=button >> text="Undo"' ).click();
```

Comment:
Same here, and we can just click on the selector using its shorthand:

```suggestion
		await snackBar.click( 'role=button[name="Undo"i]' );
```

Reply:
https://github.com/WordPress/gutenberg/pull/41310/files#r883563621: Well, then I was wrong here, we can't use the shorthand 😆 (I thought it's `page` 😅)